### ISSUE TITLE
Connection reset by peer

### ISSUE THREAD

**Issue by OmriBenShoham** [PostHog/posthog#19726]
## Bug description
I'm using Posthog client via my python flask code :

```
from posthog import Posthog
client = Posthog(api_key="SOME_API_KEY", host="SOME_POSTHOG_HOST)
client.feature_enabled(key="SOME_FEATURE_NAME", distinct_id="SOME_DISTINCT_ID")
```

The issue is that in the last few hours, I'm starting to get connection reset errors:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/posthog/client.py", line 576, in get_feature_flag
    feature_flags = self.get_feature_variants(
  File "/usr/local/lib/python3.9/site-packages/posthog/client.py", line 137, in get_feature_variants
    resp_data = self.get_decide(distinct_id, groups, person_properties, group_properties, disable_geoip)
  File "/usr/local/lib/python3.9/site-packages/posthog/client.py", line 164, in get_decide
    resp_data = decide(self.api_key, self.host, timeout=10, **request_data)
  File "/usr/local/lib/python3.9/site-packages/posthog/request.py", line 66, in decide
    res = post(api_key, host, "/decide/?v=3", gzip, timeout, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/posthog/request.py", line 41, in post
    res = _session.post(url, data=data, headers=headers, timeout=timeout)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 637, in post
    return self.request("POST", url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/ddtrace/contrib/requests/connection.py", line 63, in _wrap_request
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.9/site-packages/ddtrace/contrib/requests/connection.py", line 122, in _wrap_send
    response = func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/requests/adapters.py", line 501, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
```

## How to reproduce
 I'm not sure it's reproducible as it can be random or whether environment depended.

## Environment

PostHog Cloud US, project ID: 21473

**Comment by GalDayan** [comment]
Same here!

**Comment by neilkakkar** [comment]
Hi @OmriBenShoham @GalDayan , not seeing anything in our telemetry that would point to an issue here, is this still happening?

**Comment by GalDayan** [comment]
@neilkakkar There was quiet for two weeks. now that happens again.

**Comment by OmriBenShoham** [comment]
@neilkakkar This error causes the flag value retrieved to be NULL:
![image (3)](https://github.com/PostHog/posthog/assets/26386874/35c272c1-b09e-414d-9793-cdffc9a6019e)

**Comment by neilkakkar** [comment]
Hello, this is happening because cloudflare is rejecting requests for some reason.

For now, we have a host that bypasses cloudflare, and if you face this as well you can update your posthog-python SDK with

host='https://us-proxy-direct.i.posthog.com'  or host='https://eu-proxy-direct.i.posthog.com' depending on US/EU installation

**Comment by neilkakkar** [comment]
As an update, we've also just released posthog-python v3.4.0 which should handle this for you

**Comment by GalDayan** [comment]
Thank you so much! @neilkakkar
